homeassistant/components/squeezebox/media_player.py

```diff
@@ -182,133 +177,41 @@ async def async_service_handler(service):
-        self._password = password
-
-    async def create_players(self):
-        """Create a list of devices connected to LMS."""
-        result = []
-        data = await self.async_query("players", "status")
-        if data is False:
-            return None
-        for players in data.get("players_loop", []):
-            player = SqueezeBoxDevice(self, players["playerid"], players["name"])
-            await player.async_update()
-            result.append(player)
-        return result
-
-    async def async_query(self, *command, player=""):
-        """Abstract out the JSON-RPC connection."""
-        auth = (
-            None
-            if self._username is None
-            else aiohttp.BasicAuth(self._username, self._password)
-        )
-        url = f"http://{self.host}:{self.port}/jsonrpc.js"
-        data = json.dumps(
-            {"id": "1", "method": "slim.request", "params": [player, command]}
-        )
-
-        _LOGGER.debug("URL: %s Data: %s", url, data)
-
-        try:
-            websession = async_get_clientsession(self.hass)
-            with async_timeout.timeout(TIMEOUT):
-                response = await websession.post(url, data=data, auth=auth)
-
-                if response.status != HTTP_OK:
-                    _LOGGER.error(
-                        "Query failed, response code: %s Full message: %s",
-                        response.status,
-                        response,
-                    )
-                    return False
-
-                data = await response.json()
-
-        except (asyncio.TimeoutError, aiohttp.ClientError) as error:
-            _LOGGER.error("Failed communicating with LMS: %s", type(error))
-            return False
-
-        try:
-            return data["result"]
-        except AttributeError:
-            _LOGGER.error("Received invalid response: %s", data)
-            return False
-
-
 class SqueezeBoxDevice(MediaPlayerEntity):
-    """Representation of a SqueezeBox device."""
+    """
+    Representation of a SqueezeBox device.
+
+    Wraps a pysqueezebox.Player() object.
+    """
 
-    def __init__(self, lms, player_id, name):
+    def __init__(self, player):
         """Initialize the SqueezeBox device."""
-        super().__init__()
-        self._lms = lms
-        self._id = player_id
-        self._status = {}
-        self._name = name
+        self._player = player
         self._last_update = None
-        _LOGGER.debug("Creating SqueezeBox object: %s, %s", name, player_id)
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self._player.name
 
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._id
+        return self._player.player_id
 
     @property
     def state(self):
         """Return the state of the device."""
-        if "power" in self._status and self._status["power"] == 0:
+        if not self._player.power:
```

Comment:
I missed this last suggestion.
```suggestion
        if self._player.power is not None and not self._player.power:
```